### PR TITLE
Avoid SQL assertions

### PR DESCRIPTION
**Pull request by greg0ire**
doctrine/dbal is the component responsible for generating the queries.
Let us make the test suite more robust by asserting that things work
from a functional point of view.

### CODE REVIEW

**Comment by greg0ire** [comment]
I didn't rid the entire codebase from calls to `getLastLoggedQuery()`, but did what I could when I understood clearly what to do. Feel free to checkout the branch and propose/push additional changes.

**Review by derrabus — DISMISSED** [review]

File: tests/Doctrine/Tests/ORM/Functional/EntityRepositoryTest.php [hunk 1]
```diff
@@ -721,29 +717,45 @@ public function testInvalidOrientation(): void
...
-            self::assertSame(
-                [
-                    'sql' => 'SELECT t0.id AS id_1, t0.country AS country_2, t0.zip AS zip_3, t0.city AS city_4, t0.user_id AS user_id_5 FROM cms_addresses t0 WHERE t0.user_id IN (?, ?, ?)',
-                    'params' => [1 => 1, 2 => 2, 3 => 3],
-                    'types' => array_fill(1, 3, ParameterType::INTEGER),
-                ],
-                $this->getLastLoggedQuery()
-            );
-        }
+        $pigs = $repo->findBy(['user' => $ids]);
```
Comment (morozov) [inline tests/Doctrine/Tests/ORM/Functional/EntityRepositoryTest.php]: As far as I understand, this is `$addresses`, not `$pigs`.

File: tests/Doctrine/Tests/ORM/Functional/EntityRepositoryTest.php [hunk 2]
```diff
@@ -721,29 +717,45 @@ public function testInvalidOrientation(): void
...
+
+        self::assertCount(3, $pigs);
```
Comment (morozov) [inline tests/Doctrine/Tests/ORM/Functional/EntityRepositoryTest.php]: Given that the previous version of the test makes an assertion regarding the query parameters, here, we need to check if the same three addresses have been returned, not just three arbitrary addresses.

File: tests/Doctrine/Tests/ORM/Functional/OneToOneEagerLoadingTest.php [hunk 3]
```diff
@@ -146,24 +146,18 @@ public function testEagerLoadWithNullableColumnsGeneratesLeftJoinOnBothSides():
...
+        $this->_em->persist($driver);
         $this->_em->flush();
         $this->_em->clear();
 
         $train = $this->_em->find(get_class($train), $train->id);
-        $this->assertSQLEquals(
-            'SELECT t0.id AS id_1, t0.driver_id AS driver_id_2, t3.id AS id_4, t3.name AS name_5, t0.owner_id AS owner_id_6, t7.id AS id_8, t7.name AS name_9 FROM Train t0 LEFT JOIN TrainDriver t3 ON t0.driver_id = t3.id INNER JOIN TrainOwner t7 ON t0.owner_id = t7.id WHERE t0.id = ?',
-            $this->getLastLoggedQuery()['sql']
-        );
+        self::assertNotNull($train, 'It should be possible to find the train even though it has no driver');
```
Comment (morozov) [inline tests/Doctrine/Tests/ORM/Functional/OneToOneEagerLoadingTest.php:158]: Should we check for a specific train, not for an arbitrary one?

File: tests/Doctrine/Tests/ORM/Functional/EntityRepositoryTest.php (hunk 2 above)
Comment (greg0ire) [reply tests/Doctrine/Tests/ORM/Functional/EntityRepositoryTest.php]: Ok, I'll add a wolf with a 4th address in there.

**Review by derrabus — APPROVED** [review]

**Pull request merged**